src/core/event_engine/effect_dispatcher.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{
+    core::event_engine::{Effect, EffectHandler, EffectInvocation, Event},
+    lib::alloc::{rc::Rc, vec, vec::Vec},
+};
+use phantom_type::PhantomType;
+use spin::rwlock::RwLock;
+
+/// State machine effects dispatcher.
```

Comment:
TYI: That's nice to have docs but you're not obligated to do them to all private stuff ;d.

Reply:
I got it. It just a matter of habit.